service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/FeedbackReceiver.java

```diff
@@ -125,6 +127,21 @@ public FeedbackReceiver(String hostName, String userName, String sasToken, IotHu
         // Codes_SRS_SERVICE_SDK_JAVA_FEEDBACKRECEIVER_12_003: [The constructor shall create a new instance of AmqpReceive object]
         this.amqpReceive = new AmqpReceive(hostName, userName, sasToken, iotHubServiceClientProtocol, proxyOptions, sslContext);
     }
+
+    public FeedbackReceiver(String hostName, TokenCredential authenticationTokenProvider, CbsAuthorizationType authorizationType, IotHubServiceClientProtocol iotHubServiceClientProtocol, ProxyOptions proxyOptions, SSLContext sslContext)
```

Comment:
will TokenCredential and CbsAuthorizationType both exist for a workflow? won't TokenCredential always be for authType = rbac?

Reply:
TokenCredential will allow for tokens of Symmetric keys (what we currently support) and for tokens of RBAC keys (what we will support). The important thing here is that the user tells us what type of keys their TokenCredential will generate. When we send the CBS auth message, we have to forward both the token and the type of that token so that the service knows how to interpret the token